jvm-libs/linea/metrics/micrometer/src/main/kotlin/net/consensys/linea/metrics/micrometer/MicrometerMetricsFacade.kt

```diff
@@ -37,74 +38,64 @@ class MicrometerMetricsFacade(
 
   init {
     if (metricsPrefix != null) requireValidMicrometerName(metricsPrefix)
+    commonTags.forEach { requireValidMicrometerName(it.key) }
   }
 
-  private fun metricHandle(category: LineaMetricsCategory?, metricName: String): String {
+  private fun metricHandle(category: MetricsCategory, metricName: String): String {
     val prefixName = if (metricsPrefix == null) "" else "$metricsPrefix."
-    val categoryName = if (category == null) "" else "$category."
+    val categoryName = "${category.toValidMicrometerName()}."
```

Comment:
Nit: I think it's more logical to move the dot to the pattern string below. `categoryName` defines the name, the string defines the pattern.